tests/test_fftfit.py

```diff
@@ -0,0 +1,291 @@
+from functools import wraps
+
+import numpy as np
+import pytest
+import scipy.stats
+from hypothesis import assume, given
+from hypothesis.extra.numpy import arrays
+from hypothesis.strategies import (
+    complex_numbers,
+    composite,
+    floats,
+    fractions,
+    integers,
+    just,
+    one_of,
+)
+from numpy.testing import assert_allclose, assert_array_almost_equal
+
+import pint.profile.fftfit_aarchiba as fftfit
+from pint.profile import fftfit_aarchiba
+from pint.profile import fftfit_nustar
+
+
+def assert_rms_close(a, b, rtol=1e-8, atol=1e-8):
+    assert np.mean((a - b) ** 2) < rtol * (np.mean(a ** 2) + np.mean(b ** 2)) + atol
+
+
+def assert_allclose_phase(a, b, atol=1e-8):
+    assert np.all(np.abs(fftfit.wrap(a - b)) <= atol)
+
+
+@composite
+def powers_of_two(draw):
+    return 2 ** draw(integers(4, 16))
+
+
+@composite
+def vonmises_templates(draw, ns=powers_of_two(), phase=floats(0, 1)):
+    return fftfit.vonmises_profile(draw(floats(1, 1000)), draw(ns), draw(phase))
```

Comment:
Because there are no dependencies between the drawn values, you could use `builds()` here instead.

```suggestion
def vonmises_templates(ns=powers_of_two(), phase=floats(0, 1)):
    return builds(fftfit.vonmises_profile, floats(1, 1000), ns, phase)
```

This involves slightly less indirection, but the main benefit is readability rather than performance.